### ISSUE TITLE
Space between widgets reset from 0 to 20 after update to latest version

### ISSUE THREAD

**Issue by spoongefeesh** [elementor/elementor#12398]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x ] The issue still exists against the latest stable version of Elementor.

## Description

After update to latest version, the "Widgets space (px)" setting found under Site Settings > Layout Settings has reset from 0 (set previous to update) to 20px. This has occurred across all sites I own that use Elementor.

## Steps to reproduce

Update to latest version of Elementor.

## Isolating the problem

- [ x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x ] This bug happens with a default WordPress theme active.
- [x ] I can reproduce this bug consistently using the steps above.

## Environment

					
== Server Environment ==
	Operating System: Linux
	Software: Flywheel/4.1.0
	MySQL version: (Ubuntu) v5.6.32
	PHP Version: 7.2.31-1+ubuntu16.04.1+deb.sury.org+1
	PHP Max Input Vars: 4000
	PHP Max Post Size: 300M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.4.2
	Site URL: https://a1autoglaze.com
	Home URL: https://a1autoglaze.com
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-GB
	Timezone: Europe/London
	Debug Mode: Inactive

== Theme ==
	Name: Hello! Theme Child
	Version: 1.0.0
	Author: Elementor team
	Child Theme: Yes
	Parent Theme Name: Elementor Hello Theme
	Parent Theme Version: 1.0
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_GB
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields: Gravityforms Add-on
		Version: 1.2.1
		Author: Danny van Holten

	Advanced Custom Fields PRO
		Version: 5.9.0
		Author: Elliot Condon

	Classic Editor
		Version: 1.6
		Author: WordPress Contributors

	CPT-onomies: Easiest way to create Custom Post Types
		Version: 1.4.0
		Author: hypestudio,dejanmarkovic,freemius,nytogroup

	Elementor
		Version: 3.0.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.0.3
		Author: Elementor.com

	Gravity Forms
		Version: 2.4.20
		Author: Gravity Forms

	Imagify
		Version: 1.9.10
		Author: WP Media

	Invisible reCaptcha
		Version: 1.2.3
		Author: Mihai Chelaru

	ManageWP - Worker
		Version: 4.9.7
		Author: GoDaddy

	Members
		Version: 3.1.1
		Author: MemberPress

	Quick Bulk Post & Page Creator
		Version: 1.0.4
		Author: tyxla

	Redirection
		Version: 4.8
		Author: John Godley

	Yoast SEO
		Version: 14.9
		Author: Team Yoast

== Must-Use Plugins ==
	ManageWP - Worker Loader
		Version: 
		Author: ManageWP

== Elements Usage ==
	
	footer : 1
		image : 4
		nav-menu : 1
		text-editor : 2
	header : 1
		button : 2
		image : 1
		nav-menu : 1
	page : 1
		text-editor : 1
		theme-page-title : 1
	single : 1
		form : 1
		heading : 3
		image : 1
		text-editor : 2
	wp-post : 0
		accordion : 1
		button : 12
		form : 4
		heading : 69
		html : 11
		image : 68
		image-gallery : 4
		testimonial : 9
		text-editor : 42
		theme-page-title : 2

== Log ==
	
Log: showing 20 of 372020-05-11 09:37:49 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.9.1',
  'to' => '2.9.4',
)]
2020-06-11 12:44:45 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.8',
  'to' => '2.9.11',
)]
2020-06-17 15:46:19 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.11',
  'to' => '2.9.12',
)]
2020-07-01 11:19:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.9.4',
  'to' => '2.10.3',
)]
2020-07-01 12:08:30 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.12',
  'to' => '2.9.13',
)]
2020-09-02 11:03:09 [info] elementor::elementor_updater Started 
2020-09-02 11:03:09 [info] Elementor/Upgrades - _v_3_0_0_move_general_settings_to_kit Start  
2020-09-02 11:03:09 [info] Elementor/Upgrades - _v_3_0_0_move_general_settings_to_kit Finished 
2020-09-02 11:03:09 [info] Elementor/Upgrades - _v_3_0_0_move_default_colors_to_kit Start  
2020-09-02 11:03:09 [info] Elementor/Upgrades - _v_3_0_0_move_default_colors_to_kit Finished 
2020-09-02 11:03:09 [info] Elementor/Upgrades - _v_3_0_0_move_saved_colors_to_kit Start  
2020-09-02 11:03:10 [info] Elementor/Upgrades - _v_3_0_0_move_saved_colors_to_kit Finished 
2020-09-02 11:03:10 [info] Elementor/Upgrades - _v_3_0_0_move_default_typography_to_kit Start  
2020-09-02 11:03:10 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.13',
  'to' => '3.0.5',
)]
2020-09-02 11:03:10 [info] Elementor/Upgrades - _v_3_0_0_move_default_typography_to_kit Finished 
2020-09-02 11:03:10 [info] Elementor/Upgrades - _v_3_0_5_re_move_space_between_widgets_to_kit Start  
2020-09-02 11:03:10 [info] Elementor/Upgrades - _v_3_0_5_re_move_space_between_widgets_to_kit Finished 
2020-09-02 11:03:10 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.13',
  'to' => '3.0.5',
)]
2020-09-02 13:21:35 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.10.3',
  'to' => '3.0.2',
)]
2020-09-03 08:46:59 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.2',
  'to' => '3.0.3',
)]

PHP: showing 3 of 3PHP: 2019-10-01 09:43:46 [notice X 104837][/www/wp-content/plugins/elementor-pro/modules/forms/widgets/form.php::1302] Undefined index: button_icon [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-10-01 12:53:30 [notice X 2070][/www/wp-content/plugins/elementor/includes/widgets/accordion.php::505] Undefined index: icon_active [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-09-03 08:47:11 [notice X 1][/www/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2020-02-03 09:17:49 [error X 5][https://a1autoglaze.com/wp-content/plugins/elementor/assets/js/editor-modules.min.js?ver=2.8.5:2:48076] n.attributes[e.name].each is not a function 
JS: 2020-09-03 07:42:32 [error X 3][https://a1autoglaze.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.0.5:2:470469] Cannot read property \'document\' of undefined 

				

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

**Comment by fabianenzensperger** [comment]
Yup, that's a bug. The simple solution is to change it after the update in the settings page, but it would still be nice if it is not changed at all during the update.

**Comment by spoongefeesh** [comment]
> Yup, that's a bug. The simple solution is to change it after the update in the settings page, but it would still be nice if it is not changed at all during the update.

Yeah, that's what I've been doing, but when you have 20+ sites you have to do it on... sigh!

**Comment by fabianenzensperger** [comment]
> > Yup, that's a bug. The simple solution is to change it after the update in the settings page, but it would still be nice if it is not changed at all during the update.
> 
> Yeah, that's what I've been doing, but when you have 20+ sites you have to do it on... sigh!

Why not wait another 1-2 weeks until everything is really fixed? Surely that would save you a lot of work, wouldn't it?

**Comment by arielk** [comment]
Please go to Elementor > Tools > Version Control > Rerun Update Script > Migrate to v3.0

Let me know if it works for you

**Comment by spoongefeesh** [comment]
> Please go to Elementor > Tools > Version Control > Rerun Update Script > Migrate to v3.0
> 
> Let me know if it works for you

No, that didn't work sadly

**Comment by spoongefeesh** [comment]
> > > Yup, that's a bug. The simple solution is to change it after the update in the settings page, but it would still be nice if it is not changed at all during the update.
> > 
> > 
> > Yeah, that's what I've been doing, but when you have 20+ sites you have to do it on... sigh!
> 
> Why not wait another 1-2 weeks until everything is really fixed? Surely that would save you a lot of work, wouldn't it?

Fair point, you're probably right!

**Comment by arielk** [comment]
We have tried to reproduce the issue in other installations and so far we have not been able to recover it, so in the meantime I am closing this issue.

Thanks for your response. Feel free to reopen the issue and we'll be happy to help you further.

**Comment by fabianenzensperger** [comment]
> We have tried to reproduce the issue in other installations and so far we have not been able to recover it, so in the meantime I am closing this issue.
> 
> Thanks for your response. Feel free to reopen the issue and we'll be happy to help you further.

@arielk Um, but it's being reset on every single page after the 3.0.X update! It would be good if that got fixed. It's just a minor fact, but many are sure to wonder about it.

**Comment by arielk** [comment]
@fabianenzensperger In every single page? you can change it global > Site Settings >  Layout

**Comment by fabianenzensperger** [comment]
> @fabianenzensperger In every single page? you can change it global > Site Settings > Layout

@arielk Sorry I meant every website that I have updated so far.

**Comment by arielk** [comment]
The solution I brought to you above is supposed to address this issue, so give it a try.
In any case we will test this in some additional development environments to ensure that the issue is resolved.

**Comment by fabianenzensperger** [comment]
> The solution I brought to you above is supposed to address this issue, so give it a try.
> In any case we will test this in some additional development environments to ensure that the issue is resolved.

@arielk Yes, I always do this solution, but it's a bit annoying when it changes (resets) after the update. Wouldn't it be easier if the setting that was made from 2.9.X is not also carried over to 3.0.X? I am just asking.

**Comment by arielk** [comment]
@fabianenzensperger @spoongefeesh Sorry guys, you're right! 

I checked it again and it seems the problem still exists. So, thank you very much for your response, we will send the fix to the next version.

**Comment by fabianenzensperger** [comment]
> @fabianenzensperger @spoongefeesh Sorry guys, you're right!
> 
> I checked it again and it seems the problem still exists. So, thank you very much for your response, we will send the fix to the next version.

@arielk Thank you! I'm curious. Also on the other fixes regarding #12363 Any idea when the 3.0.8 update will come?

**Comment by arielk** [comment]
@fabianenzensperger 

The next version 3.0.8 (global fonts/colors fixes) will be release in up coming hours.
The space between fix should go into the next patch version that comes out in the coming days.
